src/libs/actions/Report.js

```diff
@@ -54,6 +56,35 @@ Onyx.connect({
     },
 });
 
+/**
+ * Map of the most recent non-loading report history items we have stored for a report. Not to be confused with the
+ * reportMaxSequenceNumbers. We can compare these values to those to determine whether there is "available history"
+ * to fetch from the server. maxSequenceNumber is retrieved when getting the reportList whereas these keep track of
+ * the actual reportActions saved in Onyx.
+ */
+const maxReportActionsSequenceNumbers = {};
```

Comment:
This is definitely confusing with `maxSequenceNumber` now and unfortunately, the comment doesn't clear it up enough. I think they are both listening to different keys, but I can't figure out why there would be a difference between the two, why that difference is OK, and what that difference has to do with history being available to download from the server. I think the comment can do a better job of explaining why each map is different, as well as explaining why they can get out of sync, why it's OK that they get out of sync, and what is useful about that information as it pertains to downloading new content from the server. If you want me to take a stab at it, let me know, but I think I need to make sure I get my head wrapped around this logic first before I can be of any more help.

Reply:
Lemme try a bit harder here. It sounds like you're getting it, but the comment doesn't explain some things that make sense to _me_ because I wrote this code.

Reply:
Just went for this! I think it works and resolves a concern I have about people getting confused about which map to refer to in the future.